### ISSUE TITLE
PUT /:courseId/team

### ISSUE THREAD

**Issue by rtholmes** [ubccpsc/classportal-backend#5]
This one is not an admin route because both admins and students can do it.

Payload:

```
{ members: [id1, id2, id3] }
```

The backend should check that the number of ids is between `minTeamSize` and `maxTeamSize` for the course if submitted by a student but should ignore those limits if submitted by an admin. What the ids are is intentionally vague here: the students will be listed by name so that might be perfect. We don't want to use student numbers (students shouldn't have access to each other's) and can't use github ids either because we don't want to require students to have associated their github ids before they are put in groups.

Is PUT because teams can't be updated (by students). POST for teams (admins only) will be deferred until later.

Important: the PUT must verify that none of the members are currently in teams. If they are, the PUT should fail.

**Comment by andrewstec** [comment]
Done based on specs, tests passing.

Just to confirm, based on what Patrice mentioned,
- The PUT verifies that none of the members are currently in a team for a particular Deliverable. Hence, members can be added to teams for a Deliverable only once.

Based on the intentionally vague naming for privacy reasons, 
- I implemented a /:courseId/students endpoint with sanitized information to retrieve the names of student who we can add to a course. The santized information is only the first name, last name, and ObjectId(), which will allow us to avoid any duplicate name issues on the front-end without revealing any CSID and SNUMs either.
-
